### DISCUSSION TITLE
Inheritance and parent object property

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#4374] [Questions]
We have written some base classes that inherit from 'BusinessBase' etc.
Our custom objects are then inherited from our base classes instead of BusinessBase (and other csla classes)
We decided to store the names of the objects making inserts/updates in the database and have a property for that in our base classes. Where should we set that property? Could we use the constructors? We do have more than one 'dataportal create' methods in most of our objects.

Kind Regards

**Comment by StefanOssendorf** [comment]
Hi. Could you provide a code snippet? It depends on how you manage those settings? 
Plain c# properties because they don't need to travel/are easily obtained or are they Csla properties?
The former constructor would be good. The latter on application context set so you didn't get null reference exceptions.

**Reply by Chicagoan2016** [reply]
They are csla properties. We just need to update those properties with something like nameof(customclass).

**Reply by StefanOssendorf** [reply]
This should be your place to go. 
https://github.com/MarimerLLC/csla/blob/b6196ca4d65c79ea4f75db6c05d653e07f4437d2/Source/Csla/Core/BusinessBase.cs#L65
But is the compile time constant worth the Csla overhead in that case?

**Comment by Chicagoan2016** [comment]
I am thinking there is an easy way : )

**Reply by StefanOssendorf** [reply]
What do you mean?
